src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Modal, Col, Button, Alert } from 'patternfly-react';
+
+import { FormFactory } from '../Form/FormFactory';
+import { createBaremetalHost } from '../../k8s/request';
+
+const formFields = {
+  name: {
+    id: 'name',
+    title: 'Name',
+    required: false,
+  },
+  controller: {
+    id: 'controller',
+    title: 'Management Controller Address',
+    required: true,
+  },
+  username: {
+    id: 'username',
+    title: 'Username',
+    required: true,
+  },
+  password: {
+    id: 'password',
+    title: 'Password',
+    type: 'password',
+    required: true,
+  },
+};
+
+export class CreateBaremetalHostDialog extends React.Component {
+  state = {
+    form: {
+      value: {},
+      valid: false,
+    },
+    isSubmitting: false,
+    errorMessage: null,
+  };
+
+  onFormChange = (newValue, target, formValid) => {
+    this.setState(state => {
+      const form = { ...state.form };
+      form.value[target] = {
+        ...newValue,
+      };
+      form.valid = formValid;
+      return { form, errorMessage: null };
+    });
+  };
+
+  onSubmit = async () => {
+    this.setState(state => ({
+      ...state,
+      isSubmitting: true,
+    }));
+
+    return createBaremetalHost(this.props.k8sCreate, this.state.form.value, this.props.selectedNamespace)
+      .then(result => {
+        this.setState(state => ({
+          ...state,
+          isSubmitting: false,
+        }));
+        this.props.onClose();
+        return result;
+      })
+      .catch(({ message }) => {
+        this.setState(state => ({
+          ...state,
+          isSubmitting: false,
+          errorMessage: message,
+        }));
+      });
+  };
+
+  render() {
+    return (
+      <Modal show bsSize="large" onHide={this.props.onClose}>
+        <Modal.Header closeButton>
+          <Modal.Title>Add Host</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <Col sm={12}>
+            <p>
+              Specify the host&apos;s name, management controller IP address, and credentials to add it to the cluster.
+            </p>
+            <FormFactory
+              fields={formFields}
+              fieldsValues={this.state.form.value}
+              onFormChange={this.onFormChange}
+              horizontal={false}
+            />
+          </Col>
+        </Modal.Body>
+        <Modal.Footer>
+          {this.state.errorMessage && <Alert type="error">{this.state.errorMessage}</Alert>}
+          {this.state.isSubmitting && <span className="spinner spinner-xs spinner-inline" />}
```

Comment:
please use InlineLoading component and customize it or create new one to your liking to include this use-case